### DISCUSSION TITLE
Grid visualization using custom images

### DISCUSSION THREAD

**Discussion by rushikumark** [mesa/mesa#1672] [Q&A]
I have recently started a project on ABM using mesa. I was able to draw a grid and setup the agent portrayal method with shapes like circle or rectangle and initiate the server to see the grid visualization (mesa.visualization.CanvasGrid) but, facing issues with setting an image as an agent to visualize in the grid. I checked the documentation related to visualization but, found nothing helpful. Please help me with any approach you know. It's very urgent. Thanks in advance.

**Comment by reka** [comment]
## Example in the mesa-examples Repo

An example using custom images is `wolf_sheep` in the `mesa-examples` repo:
https://github.com/projectmesa/mesa-examples/tree/main/examples/wolf_sheep

### Some things to note:

1. The images are located in the directory `wolf_sheep/wolf_sheep/resources`
https://github.com/projectmesa/mesa-examples/tree/main/examples/wolf_sheep/wolf_sheep/resources
2. The images are visible in the browser, only if you start the server from the "outer" `wolf_sheep` directory with `mesa runserver`.
If you run `mesa runserver <directory_path>` from any other location:

* The simulation is running as expected. :heavy_check_mark: 
* The images aren't shown. :red_circle: 

### Example Cases

Let's assume you've cloned the repo mesa-examples into the directory `~/code`.

#### Case: No Images

```
cd
mesa runserver code/mesa-examples/examples/wolf_sheep
```

=>

* The simulation is running. :heavy_check_mark: 
* No images are shown. :red_circle: 
* You can see 404 messages in the CLI output at each step. :notebook: 

> 404 GET /local/custom/wolf_sheep/resources/sheep.png (127.0.0.1) 1.05ms
> 404 GET /local/custom/wolf_sheep/resources/wolf.png (127.0.0.1) 0.52ms

#### Case: Images Are Shown

```
cd ~/code/mesa-examples/examples/wolf_sheep
mesa runserver
```

=>

The simulation is running. :heavy_check_mark: 
The custom sheep and wolf images are shown. :heavy_check_mark:
